### DISCUSSION TITLE
Don't delete/overwrite existing files!

### DISCUSSION THREAD

**Discussion by alexcristd** [osmandapp/OsmAnd#16287] [Ideas]
I played around with the location of the files (Android 13, Samsung One UI 5.0), trying to find a place where I can access them from the computer.

I finally tried a location where OsmAnd didn't have permission (/Android/com), it failed to move the existing files, showed an error and asked to restart the app. After restart, the file location was reverted to the previous one (/Android/obb) and EVERYTHING was GONE. I don't care about maps, I can always download them. But I effectively lost 1 year of track recordings!

For the love of God and all that's holy in the Universe, DON'T DO THIS! Please! Make sure you don't overwrite/remove existig files without at least showing a warning or something! Geeeeezzz... 😢

**Comment by pebogufi** [comment]
"Playing around with the location of the files" without having made a backup... 😵‍
DON'T DO THIS!

**Reply by ThomasMeister123** [reply]
Its so easy to backup and restore WITHIN the app! Sorry, no pity...

**Reply by alexcristd** [reply]
Really? The backup folder is between the same folders that were erased. Just saying... 

On the other hand, any alternatives don't excuse a bad design to start with. Removing files with no warning is a big no-no.

As far as making a manual copy of the files, it's exactly what I was trying to do. But the freaking idiotic restrictions in Android 13 don't allow the user to see all files, which is why I was playing with that option in the first place, trying to move them somewhere I could access them from the computer. In previous Android versions I could copy them from the original location, but that was too easy...

**Reply by tommycrock** [reply]
Whether or not osmand has an easy backup option it should behave gracefully. Nothing wrong with reporting this as an issue.

**Comment by vshcherb** [comment]
Files are rarely GONE, it's not by design. They left at previous location, so if you have 2 locations files probably exist at 2 locations at the same time. So in the end you can make backup and change it again. 
Files are deleted only if they are moved or fully copied, so OsmAnd always maintains 1 copy of the files

**Reply by alexcristd** [reply]
Well, sort of. Here are more details:

All files were moved successfully a couple times (so it does work, no comment there). They ended up in /Android/obb, everything was fine.

Finly tried to move them in /Android/com. It DID create a full folder structure successfully. 25 folders in total. Then it threw an error while copying the first file (I think it was a map), and asked for a restart of the app - just like it does after a successful change of location.

After the app restarted, I still have the full folder structure in /com - all 25 EMPTY folders. And the app reverted to using the ones in /obb - but now everything in there is also empty.

It's not only the recordings that are gone, but everything. Maps, settings, all of it. It just behaved as if it was the first time I opened the app after installation. 

When and why the files were removed, I don't know. If there's any default log somewhere that I can provide, I'll gladly do it. Otherwise, I can only give you my word.

**Reply by ThomasMeister123** [reply]
"Backup" is NOT just save the Data in original folder. If these data are soooo Important for you, you can read this:
https://www.uschamber.com/co/co/run/technology/3-2-1-backup-rule#:~:text=Here%27s%20what%20the%203%2D2,least%20one%20backup%20file%20offsite.
